src/widget/datetime/datetimepicker-extended.js

```diff
@@ -168,7 +168,11 @@ class DatetimepickerExtended extends Widget {
         const val = value ? new Date( value ).toISOLocalString() : '';
         const vals = val.split( 'T' );
         const dateVal = vals[ 0 ];
-        const timeVal = ( vals[ 1 ] && vals[ 1 ].length > 4 ) ? vals[ 1 ].substring( 0, 5 ) : '';
+        /**
+         * seems the source of issue #649 is in the toISOLocalString function 
+         * refer: https://github.com/enketo/enketo-xpathjs/blob/master/src/date-extensions.js#L16
```

Comment:
Yes, I think you're right. That is not a great trick (in date-extensions.js) but I don't think we can remove it atm.

Reply:
It was introduced here: https://github.com/enketo/enketo-xpathjs/commit/5dbfa2107a9ca9efe072608bc1201b1c927d37bf#diff-eb47f20ec88abc06d935a1c5511de69f